x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,89 +21,103 @@ export class JsonIndexFilePicker extends Component {
   state = {
     fileUploadError: '',
     fileParsingProgress: '',
-    fileRef: null
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.fileRef !== this.props.fileRef) {
-      this.setState({ fileRef: this.props.fileRef });
-    }
+  async componentDidMount() {
+    this._isMounted = true;
   }
 
-  _fileHandler = async fileList => {
-    const {
-      resetFileAndIndexSettings, setParsedFile, onFileRemove, onFileUpload,
-      transformDetails, setFileRef, setIndexName
-    } = this.props;
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
 
-    const { fileRef } = this.state;
+  _fileHandler = async fileList => {
```

Comment:
Why is this method async? The caller does not handle any async response and the function returns nothing